geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/SScanExecutor.java

```diff
@@ -108,53 +99,17 @@ public void executeCommand(Command command, ExecutionHandlerContext context) {
       return;
     }
 
-    @SuppressWarnings("unchecked")
-    List<ByteArrayWrapper> returnList =
-        (List<ByteArrayWrapper>) getIteration(new ArrayList<>(set), matchPattern,
-            count, cursor);
-
+    RedisSetCommands redisSetCommands =
+        new RedisSetCommandsFunctionExecutor(context.getRegionProvider().getSetRegion());
+    List<Object> returnList = redisSetCommands.sscan(key, matchPattern, count, cursor);
+    if (returnList == null) {
+      returnList = Collections.emptyList();
+    }
```

Comment:
I don't think `returnList` can ever be null.  In that case, this check is unnecessary

Reply:
done